source/Calamari/Scripts/Octopus.Features.IISWebSite_BeforePostDeploy.ps1

```diff
@@ -481,8 +481,9 @@ if ($deployAsWebSite)
 		}
 
 		Write-Host "Finding SSL certificate with thumbprint $sslCertificateThumbprint"
-	
-		$certificate = Get-ChildItem Cert:\LocalMachine -Recurse | Where-Object { $_.Thumbprint -eq $sslCertificateThumbprint -and $_.HasPrivateKey -eq $true } | Select-Object -first 1
+
+		$accessibleCertificates = Get-ChildItem Cert:\LocalMachine | Foreach-Object { Get-ChildItem "Cert:\LocalMachine\$($_.Name)" -ErrorAction Ignore }
```

Comment:
Did you consider `-ErrorAction SilentlyContinue`?
Either that, or changing it to a loop, rather than chaining, and then we can have some nicer logging?

something like this (code untested):
```
foreach ($certStore in (Get-ChildItem Cert:\LocalMachine)) {
    $certs = Get-ChildItem "Cert:\LocalMachine\$($certStore.Name)" -ErrorAction SilentlyContinue -ErrorVariable $certError
    if ($certError) {
        Write-Warning "Unable to access certificate store $($certStore.Name): $certError"
    } else {
        $certificate = $certs | Where-Object { $_.Thumbprint -eq $sslCertificateThumbprint -and $_.HasPrivateKey -eq $true } | Select-Object -first 1
    }
}
```